src/settingsholder.h

```diff
@@ -131,13 +131,14 @@ class SettingsHolder final : public QObject {
   GETSET(bool, hasConnectionChangeNotification, connectionChangeNotification,
          setConnectionChangeNotification);
   GETSET(bool, hasFeaturesTourShown, featuresTourShown, setFeaturesTourShown);
-
   GETSET(QStringList, hasMissingApps, missingApps, setMissingApps)
   GETSET(QStringList, hasSeenFeatures, seenFeatures, setSeenFeatures)
   GETSET(QStringList, hasDevModeFeatureFlags, devModeFeatureFlags,
          setDevModeFeatureFlags);
   GETSET(QStringList, hasRecentConnections, recentConnections,
          setRecentConnections);
+  GETSET(bool, hasDeviceKeyRegenerated_2_5, deviceKeyRegenerated_2_5,
```

Comment:
Could we instead do this by storing the version number under which the key current device key was generated? That way we don't have to keep adding bespoke settings every time an issue like this occurs and the question instead becomes a comparison of version numbers.

Reply:
For example: `if (settingsHolder->deviceKeyVersion() < "2.5") { /* do some stuff */ }`